tracking/translations_parser/parser.py

```diff
@@ -215,6 +223,100 @@ def _join(seq):
 
             yield headers, text
 
+    def get_extra_marian_config(self) -> dict:
+        """
+        Read extra configuration files (Marian, OpusTrainer, extra CLI arguments).
+        Publication outside of a Taskcluster context (offline mode) cannot access
+        the configuration files, only extra-args will be set in this case.
+        """
+        extra_config = {
+            "extra-args": None,
+            "model": None,
+            "training": None,
+            "datasets": None,
+            "opustrainer": None,
+        }
+
+        if (
+            self.description is None
+            or (match := MARIAN_ARGS_REGEX.search(self.description)) is None
+        ):
+            logger.error(self.description)
+            logger.warning(
+                "Invalid Marian description, skipping Marian and OpusTrainer configuration detection."
+            )
+            return extra_config
+
+        logger.info("Reading Marian command line arguments.")
+        (arguments_str,) = match.groups()
+        # Build args from the command line input text
+        args = defaultdict(list)
+        key = None
+        for i in iter(shlex.split(arguments_str)):
+            if i.startswith("-"):
+                key = i.strip("-")
+                continue
+            args[key].append(i)
+
+        # Handle extra arguments used to run Marian
+        extra_config["extra-args"] = {}
+        for arg, values in list(args.items())[::-1]:
+            if arg == LAST_MARIAN_DECLARED_ARGUMENT:
+                break
+            # Do not use lists for single values
+            if len(values) == 1:
+                extra_config["extra-args"].update({arg: values[0]})
+            else:
+                extra_config["extra-args"].update({arg: values})
+
+        if os.environ.get("TASK_ID") is None:
+            logger.info(
+                "Extra configuration files can only be retrieved in Taskcluster context, skipping."
+            )
+            return extra_config
+
+        # Handle Marian model and training YAML configuration files
+        if len(args["c"]) != 2:
+            logger.warning("Invalid")
+        else:
+            model_path, training_path = args["c"]
```

Comment:
the order is not guaranteed here. We can identify which config is which based on their names